fink_filters/filter_rate_based_kn_candidates/filter.py

```diff
@@ -0,0 +1,305 @@
+            """.format(dict_filt[fid[i]], mag, err_mag, rate, error_rate)
+        radec_text = """
+             *RA/Dec:*\n- [hours, deg]: {} {}\n- [deg, deg]: {:.7f} {:+.7f}
+             """.format(ra_formatted[i], dec_formatted[i], ra[i], dec[i])
+        galactic_position_text = """
+            *Galactic latitude:*\n- [deg]: {:.7f}""".format(b[i])
+
+        tns_text = '*TNS:* <https://www.wis-tns.org/search?ra={}&decl={}&radius=5&coords_unit=arcsec|link>'.format(ra[i], dec[i])
+        # message formatting
+        blocks = [
+            {
+                "type": "section",
+                "fields": [
+                    {
+                        "type": "mrkdwn",
+                        "text": alert_text
+                    },
+                ]
+             },
+            {
+                "type": "section",
+                "fields": [
+                    {
+                        "type": "mrkdwn",
+                        "text": time_text
+                    },
+                    {
+                        "type": "mrkdwn",
+                        "text": score_text
+                    },
+                    {
+                        "type": "mrkdwn",
+                        "text": radec_text
+                    },
+                    {
+                        "type": "mrkdwn",
+                        "text": measurements_text
+                    },
+                    {
+                        "type": "mrkdwn",
+                        "text": galactic_position_text
+                    },
+                    {
+                        "type": "mrkdwn",
+                        "text": tns_text
+                    },
+                ]
+            },
+        ]
+
+        error_message = """
+        {} is not defined as env variable
+        if an alert has passed the filter,
+        the message has not been sent to Slack
+        """
+        for url_name in ['KNWEBHOOK', 'KNWEBHOOK_FINK']:
+            if (url_name in os.environ):
+                requests.post(
+                    os.environ[url_name],
+                    json={
+                        'blocks': blocks,
+                        'username': 'Rate-based kilonova bot'
+                    },
+                    headers={'Content-Type': 'application/json'},
+                )
+            else:
+                log = logging.Logger('Kilonova filter')
+                log.warning(error_message.format(url_name))
+
+        ama_in_env = ('KNWEBHOOK_AMA_CL' in os.environ)
+
+        # Send alerts to amateurs only on Friday
+        now = datetime.datetime.utcnow()
+
+        # Monday is 1 and Sunday is 7
+        is_friday = (now.isoweekday() == 5)
+
+        if (np.abs(b[i]) > 20) & (mag < 20) & is_friday & ama_in_env:
+            requests.post(
+                os.environ['KNWEBHOOK_AMA_RATE'],
+                json={
+                    'blocks': blocks,
+                    'username': 'kilonova bot'
```

Comment:
`username` should be `Rate-based kilonova bot`?

Reply:
Even though it is not used yet ;-)

Reply:
I can change that, for now they are all named "kilonova bot" in the amateur workspace because they are in different channels anyway - but it wouldn't hurt to repeat.